Textfield component

rolandoldengarm opened the pull request:
Updated after issue 37.
https://github.com/ngOfficeUIFabric/ng-officeuifabric/issues/37

Commit:
Textfield initial checkin

Includes a full working demo.

Todo: more examples
Todo: add more documentation

Commit:
Updated textfield

Documentation, vet issues, examples

Commit:
We don't need ts.cmd

andrewconnell commented:
Nice work! Bit of feedback... mind making a few doc-like tweaks?
- In the JSDOC for the directive in `textFieldDirective.ts`...
  - [ ] Can you remove the [first @see](https://github.com/rolandoldengarm/ng-officeuifabric/blob/textfield/src/components/textfield/textFieldDirective.ts#L15) comment? That was from the icon and not relevant to your directive.
  - [ ] Can you move the comment to before the directive declaration? Right now it's sort of linked up to the scope which doesn't make sense based on the order of where it is.
  - [ ] Can you build a JSDOC for the [scope interface](https://github.com/rolandoldengarm/ng-officeuifabric/blob/textfield/src/components/textfield/textFieldDirective.ts#L28-L36) you defined? This would help in explaining all the public properties / attributes people can use and will be used in building the docs.
- [ ] The [index.js](https://github.com/rolandoldengarm/ng-officeuifabric/blob/textfield/src/components/textfield/demo/index.js) in the demo is for the icon, not for your directive... needs to get removed.
- Can you update the [index.html](https://github.com/rolandoldengarm/ng-officeuifabric/blob/textfield/src/components/textfield/demo/index.html) for the demo... 
  - [ ] the [title refers to the icon](https://github.com/rolandoldengarm/ng-officeuifabric/blob/textfield/src/components/textfield/demo/index.html#L5)
  - [ ] the [icon sample is still in the code but commented out](https://github.com/rolandoldengarm/ng-officeuifabric/blob/textfield/src/components/textfield/demo/index.html#L50-L66)... it really shouldn't be there. 
  - [ ] the [H1 on the page also refers to the icon directive](https://github.com/rolandoldengarm/ng-officeuifabric/blob/textfield/src/components/textfield/demo/index.html#L20)

Ideally would be good to see the coverage for the `textFieldDirective.ts` file up [higher than 77%](https://coveralls.io/builds/4630205/source?filename=src%2Fcomponents%2Ftextfield%2FtextFieldDirective.js)

Once you've done these updates and push them, we can get it merged in

Commit:
Update to textfield after comments

- Updated documentation
- Additional examples
- Removed references to icon (copy pasted)

ghost commented:
Pushed :) Added some unit tests (for the UI events like blur and click for placeholder). Looks a bit better now :)

andrewconnell commented:
Nice work! Just pushed it to `dev`. I took the liberty to tweak some stuff in your JSDOC comment for the `ITextFieldScope` interface if you want to see what I did for future ref:
- added type declaration to each property
- lined properties up for readability
- removed comments such as "following are properties, everything else is internal" as that's implied when they are listed as properties 
- moved the remark about `label` & `placeholder` to the description as it was going to get left out of any dynamically generated docs

You're also benefiting from the changes made from issue #56... coverage went way up. Noticed there were two paths that aren't covered in your tests (`inputFocus` & `inputBlur`)... you can see those in the local coverage report after running a test. Not big stuff... just FYI.
